lib/manager/flux/extract.ts

```diff
@@ -0,0 +1,116 @@
+import { loadAll } from 'js-yaml';
+import { HelmDatasource } from '../../datasource/helm';
+import { logger } from '../../logger';
+import { SkipReason } from '../../types';
+import { readLocalFile } from '../../util/fs';
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+import type {
+  FluxManifest,
+  FluxResource,
+  HelmRelease,
+  HelmRepository,
+} from './types';
+
+function readManifest(content: string): FluxManifest {
+  const manifest: FluxManifest = { releases: [], repositories: [] };
+  let resources: FluxResource[];
+  try {
+    resources = loadAll(content, null, { json: true }) as FluxResource[];
+  } catch (err) {
+    logger.debug({ err }, 'Failed to parse Flux manifest');
+    return null;
+  }
+
+  // It's possible there are other non-Flux HelmRelease/HelmRepository CRs out there, so we filter based on apiVersion.
+  for (const resource of resources) {
```

Comment:
Could be good to check `is.nonEmptyArray(resources)` before this, because it may not even be an array. And if it's an empty array, we may as well return null too.

Reply:
I think we're fine here, since according to the `js-yaml` docs, `loadAll` always returns an array (with one element in the common case of a doc with a single resource), and empty arrays are handled later in the flow.